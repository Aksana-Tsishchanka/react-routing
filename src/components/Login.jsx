import React from 'react';
import { browserHistory } from 'react-router';
import sendRequest from './api';

export default class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      login: '',
      password: '',
    };
    this.onChangeInput = this.onChangeInput.bind(this);
    this.onLogin = this.onLogin.bind(this);
  }


  onChangeInput({ target: { value, name } }) {
    this.setState({
      [name]: value,
    });
  }

  onLogin(e) {
    e.preventDefault();
    const init = {
      method: 'POST',
      body: JSON.stringify(this.state),
    };

    sendRequest('login', init)
      .then(response => response);
    browserHistory.push('todo');
  }

  render() {
    return (
      <div>
        <h1>Login page...</h1>
        <form>
          <div>
            <label htmlFor="login">User name:</label>
            <input
              type="text"
              name="login"
              value={this.state.login}
              onChange={this.onChangeInput}
            />
          </div>
          <div>
            <label htmlFor="password">Password:</label>
            <input
              type="text"
              name="password"
              value={this.state.password}
              onChange={this.onChangeInput}
            />
          </div>
          <button onClick={this.onLogin}>Log in</button>
        </form>
      </div>
    );
  }
}
