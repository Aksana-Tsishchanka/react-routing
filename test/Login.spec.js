import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Login from '../src/components/Login';

jest.mock('./__mock__/sendRequest');


test('Login render', () => {
  const component = renderer.create(<Login />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('onLogin has to change localStorage', () => {
  const wrapper = shallow(
    <Login />);
  wrapper.setState({
    login: '1',
    password: '1',
  });
  wrapper.instance().onLogin({ preventDefault: () => {} });
  // wait that console.log(1) will be written
});

