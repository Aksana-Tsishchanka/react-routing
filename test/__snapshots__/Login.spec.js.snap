// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login render 1`] = `
<div>
  <h1>
    Login page...
  </h1>
  <form>
    <div>
      <label
        htmlFor="login"
      >
        User name:
      </label>
      <input
        name="login"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        name="password"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      onClick={[Function]}
    >
      Log in
    </button>
    
  </form>
</div>
`;
